util/metadata_cleanup.py

```diff
@@ -105,17 +106,115 @@ def dissection_check(path, dh, sub_id, expt_date, columns):
             print("\tDissection Time: Match!")
             return
 
+def project_check(dh, sub_id, expt_date, species):
+    mouse_prod = datetime.date(2017, 10, 01)
+    project = dh.info().get('project', None)
+    if project is None:
+        if species is None and sub_id is None:
+            print("\tNo specimen, can't set project code")
+            return
+        if species.lower() == 'human':
+            print("\tSet Project Code: human course matrix")
+            return
+        if expt_date < mouse_prod:
+            print("\tSet Project Code: mouse V1 pre-production")
+        else:
+            print("\tSet Project Code: mouse V1 course matrix")
+
+def region_check(dh, species):
+    region = dh.info().get('target_region', None)
+    if species is None:
+        print("\tCan't set target region")
+        return
+    if species == 'mouse':
+        if region is None:
+            print("Set target region: V1")
+            return
+        if region == 'V1':
+            print("\tTarget region: Match!")
+        else:
+            print("\tTarget region mismatch: %s != V1" % region)
+
+def internal_check(dh, species):
+    internal = dh.info().get('internal', None)
+    internal_dye = dh.info().get('internal_dye', None)
+    if internal is None:
+        print("\tSet Internal: Standard K-Gluc")
+        return
+    if internal.lower() != 'standard k-gluc':
+        print("\tInternal mismatch: %s != Standard K-Gluc" % internal)
+        return
+    if internal_dye is not None:
+        return
+    if internal_dye is None and species is None:
+        print("\tCan't set internal dye")
+        return
+    if internal_dye is None and species.lower() == 'human':
+        print("\tSet internal dye: AF488")
+        return
+    if internal_dye is None and species != 'mouse':
+        genotype = species.split(';')
+        if len(genotype) < 3:
+            print("\tSet internal dye: AF488, %s looks likes single transgenic" % species)
+        else:
+            print("\tSet internal dye: Cascade Blue, %s looks liked quad" % species)
+    else:
+        print("\tCan't parse genotype %s, set internal dye manually" % species)
+
+def rig_check(dh):
```

Comment:
I'm confused about the logic here.. maybe:

    def rig_check(dh):
        rig = dh.info().get('rig_name', None)
        if rig in (None, ''):
            #dh.setInfo(rig_name=config.rig_name)
            print("\tRig set: %s" % config.rig_name)
            return
        elif rig != config.rig_name:
            print("\tRig mismatch: %s != %s" % (rig, config.rig_name))

Reply:
do you want the `elif rig != config.rig_name` just to be explicit? as opposed to just leaving it as an `else`, which would also catch any unforeseen weirdness

Reply:
I think the only other possibility would be `rig == config.rig_name`, which can (and should) be safely ignored.